Can't import Postman collection (exported from Revolut merchant API)

### I have checked the following:

- [x] I have searched existing issues and found nothing related to my issue.

### This bug is:

- [ ] making Bruno unusable for me
- [x] slowing me down but I'm able to continue working
- [x] annoying
- [ ] this feature was working in a previous version but is broken in the current release.

### Bruno version

2.5.0

### Operating System

MacOS 14.7.2

### Describe the bug

When importing the Postman collection file : [Merchant API.postman_collection.json](https://github.com/user-attachments/files/20745573/Merchant.API.postman_collection.json)
 , Bruno fails to import it and restarts the import process in a loop at the "select the type of your existing collection" dialog box.

In the Chrome console, I get the following error:
`index.fed3d323.js:5 Error converting Postman to Bruno via Electron: Error: Error invoking remote method 'renderer:convert-postman-to-bruno': Error: Import collection failed: Cannot read properties of undefined (reading 'replace')
(anonymous) @ index.fed3d323.js:5
Promise.catch
(anonymous) @ index.fed3d323.js:5
h6 @ index.fed3d323.js:5
h2 @ index.fed3d323.js:5
(anonymous) @ index.fed3d323.js:5
Promise.then
(anonymous) @ index.fed3d323.js:5
h5 @ index.fed3d323.js:5
onClick @ index.fed3d323.js:6
uU @ lib-react.41929aa2.js:2
(anonymous) @ lib-react.41929aa2.js:2
tC @ lib-react.41929aa2.js:2
uW @ lib-react.41929aa2.js:2
sK @ lib-react.41929aa2.js:2
sQ @ lib-react.41929aa2.js:2
index.fed3d323.js:5 Error while importing collection 0: BrunoError: Conversion failed
    at index.fed3d323.js:5:105190`

### .bru file to reproduce the bug

_No response_

### Screenshots/Live demo link

<img width="1553" alt="Image" src="https://github.com/user-attachments/assets/9ab17f0f-5522-4070-880c-e7eab573c28e" />

--- a/packages/bruno-converters/src/postman/postman-env-to-bruno-env.js
+++ b/packages/bruno-converters/src/postman/postman-env-to-bruno-env.js
@@ -6,10 +6,10 @@ const isSecret = (type) => {
   return type === 'secret';
 };
 
-const importPostmanEnvironmentVariables = (brunoEnvironment, values) => {
+const importPostmanEnvironmentVariables = (brunoEnvironment, values = []) => {
   brunoEnvironment.variables = brunoEnvironment.variables || [];
 
-  each(values, (i) => {
+  each(values.filter(i => !(i.key == null && i.value == null)), (i) => {
     const brunoEnvironmentVariable = {
       uid: uuid(),
       name: (i.key ?? '').replace(invalidVariableCharacterRegex, '_'),

--- a/packages/bruno-converters/src/postman/postman-to-bruno.js
+++ b/packages/bruno-converters/src/postman/postman-to-bruno.js
@@ -119,7 +119,7 @@ const importScriptsFromEvents = (events, requestObject) => {
 };
 
 const importCollectionLevelVariables = (variables, requestObject) => {
-  const vars = variables.map((v) => ({
+  const vars = variables.filter(v => !(v.key == null && v.value == null)).map((v) => ({
     uid: uuid(),
     name: (v.key ?? '').replace(invalidVariableCharacterRegex, '_'),
     value: v.value ?? '',

--- a/packages/bruno-converters/tests/postman/postman-env-to-bruno-env.spec.js
+++ b/packages/bruno-converters/tests/postman/postman-env-to-bruno-env.spec.js
@@ -100,13 +100,6 @@ describe('postmanToBrunoEnvironment Function', () => {
           enabled: true,
           secret: false,
           uid: "mockeduuidvalue123456",
-        },
-        {
-          name: '',
-          value: '',
-          enabled: true,
-          secret: false,
-          uid: "mockeduuidvalue123456",
         }
       ],
     };
@@ -132,5 +125,24 @@ describe('postmanToBrunoEnvironment Function', () => {
     await expect(postmanToBrunoEnvironment(invalidBrunoEnvironment)).rejects.toThrow(
       'Unable to parse the postman environment json file'
     );
+  });
+
+  it("should handle empty variables", async () => {
+    const collectionWithEmptyVars = {
+      "name": "My Environment",
+      "values": []
+    };
+
+    const brunoCollection = await postmanToBrunoEnvironment(collectionWithEmptyVars);
+    expect(brunoCollection.variables).toEqual([]);
+  });
+
+  it("should handle undefined variables", async () => {
+    const collectionWithUndefinedVars = {
+      "name": "My Environment",
+    };
+
+    const brunoCollection = await postmanToBrunoEnvironment(collectionWithUndefinedVars);
+    expect(brunoCollection.variables).toEqual([]);
   });
 });

--- a/packages/bruno-converters/tests/postman/postman-to-bruno/postman-to-bruno.spec.js
+++ b/packages/bruno-converters/tests/postman/postman-to-bruno/postman-to-bruno.spec.js
@@ -55,14 +55,23 @@ describe('postman-collection', () => {
         name: '',
         value: '',
         enabled: true
-      },
-      {
-        uid: "mockeduuidvalue123456",
-        name: '',
-        value: '',
-        enabled: true
       }
     ]);
+  });
+
+  it("should handle empty variables", async () => {
+    const collectionWithEmptyVars = {
+      "info": {
+        "_postman_id": "7f91bbd8-cb97-41ac-8d0b-e1fcd8bb4ce9",
+        "name": "collection with falsy vars",
+        "schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json"
+      },
+      "variable": [],
+      "item": []
+    };
+
+    const brunoCollection = await postmanToBruno(collectionWithEmptyVars);
+    expect(brunoCollection.root.request.vars.req).toEqual([]);
   });
 });
 

